python/tchannel/tornado/stream.py

```diff
@@ -98,11 +98,17 @@ def __init__(self, buf=None, auto_close=True):
         self._condition = Condition()
         self.auto_close = auto_close
 
+        self.err_future = tornado.gen.Future()
```

Comment:
Why are you using a future to track this. You can just track the value inside an instance variable and raise that.